GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs

```diff
@@ -13,15 +13,21 @@ public interface IUserRepositoriesListController
         Task AssignCategoryAsync(Repository repository, string? category);
         string GetCurrentBranchName(string path);
         bool IsValidGitWorkingDir(string path);
-        (IReadOnlyList<RecentRepoInfo> recentRepositories, IReadOnlyList<RecentRepoInfo> favouriteRepositories) PreRenderRepositories(Graphics g);
+        (IReadOnlyList<RecentRepoInfo> recentRepositories, IReadOnlyList<RecentRepoInfo> favouriteRepositories) PreRenderRepositories(Graphics g, string filter);
         bool RemoveInvalidRepository(string path);
+        void ClearCache();
     }
 
     public sealed class UserRepositoriesListController : IUserRepositoriesListController
     {
         private readonly ILocalRepositoryManager _localRepositoryManager;
         private readonly IInvalidRepositoryRemover _invalidRepositoryRemover;
 
+        // Holds the raw, unfiltered list of repositories.
+        // This is done to allow fast filtering of all known repos.
+        private IList<Repository>? _AllRecentRepositories;
+        private IList<Repository>? _AllFavoriteRepositories;
```

Comment:
```suggestion
        private IReadOnlyList<Repository>? _allRecentRepositories;
        private IReadOnlyList<Repository>? _allFavoriteRepositories;
```

Reply:
The `LoadFavourite/RecentHistoryAsync` calls return an `IList` not an `IReadOnlyList`

Reply:
This is a covariant conversion (I think this is the right word), i.e. you can go from `List<T>` to `IReadOnlyList<T>`.

Reply:
Yes you're right for `List<T>` but the existing methods return an `IList<T>` which does not implement `IReadOnlyList<T>`